src/server/index.js

```diff
@@ -66,6 +67,11 @@ app.prepare()
             })
         );
 
+        logger.info("adding the /login/* handler");
+        server.get("/login/*", ensureLoggedIn("/login"), (req, res) => {
```

Comment:
From what I can tell, this should be doable.  Looks like `next/router` has an `asPath` method that should return the current page.  So from the UI side, we'd be navigating to `/login/${router.asPath}`.